K/kalign/build_tarballs.jl

```diff
@@ -0,0 +1,63 @@
+using BinaryBuilder, Pkg
+
+name = "kalign"
+version = v"3.3.5"
+# url = "https://github.com/TimoLassmann/kalign"
+# description = "A fast multiple sequence alignment program."
+
+# NOTES
+# - fails to build on windows
+#   - undefined references to localtime_r
+#   - warning: implicit definition of getline
+
+sources = [
+    GitSource("https://github.com/TimoLassmann/kalign",
+              "58ca06a51b53d76d3fb96ef335fbc7110c36cd46"),
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir/kalign*/
+mkdir build && cd build
+
+# avoid autodetection (because we are cross-compiling)
+if [[ "${target}" == x86_64-* || "${target}" == i686-* ]]; then
+    CMAKE_EXTRA_FLAGS="-DHAVE_SSE=ON -DHAVE_AVX=ON -DHAVE_AVX2=ON"
+else
+    CMAKE_EXTRA_FLAGS="-DHAVE_SSE=OFF -DHAVE_AVX=OFF -DHAVE_AVX2=OFF"
+fi
+echo "CMAKE_EXTRA_FLAGS=$CMAKE_EXTRA_FLAGS"
+
+cmake .. \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=RELEASE \
+    -DBUILD_SHARED_LIBS=ON \
+    -DBUILD_TESTING=OFF \
+    -DUSE_OPENMP=ON \
+    ${CMAKE_EXTRA_FLAGS}
+make -j${nproc}
+make install
+install_license ../COPYING
+"""
+
+platforms = supported_platforms(; exclude = p -> Sys.iswindows(p))
+platforms = expand_cxxstring_abis(platforms; skip = p -> Sys.isfreebsd(p) || (Sys.isapple(p) && arch(p) == "aarch64"))
```

Comment:
Uh?  The filter doesn't look right, [nor new](https://github.com/JuliaPackaging/Yggdrasil/pull/6414#discussion_r1141382275)

Reply:
If you're always copying these wrong lines from https://github.com/JuliaPackaging/Yggdrasil/blob/9b217bae76abe35924baa185fae749c0eeddcf5c/F/FAMSA/build_tarballs.jl#L79 well, it's wrong also there, so please remove it before further spreading this error.

Reply:
As is wrong in https://github.com/JuliaPackaging/Yggdrasil/blob/9b217bae76abe35924baa185fae749c0eeddcf5c/F/Foldseek/build_tarballs.jl#L55

Reply:
And in https://github.com/JuliaPackaging/Yggdrasil/blob/9b217bae76abe35924baa185fae749c0eeddcf5c/M/MMseqs2/build_tarballs.jl#L75

Reply:
Sorry again... Will fix the other ones as well in another PR

Is there a place I can add a check so that my future self can't make that mistake again?

Reply:
https://github.com/JuliaPackaging/Yggdrasil/pull/6414#discussion_r1141419452
> You need to expand if and only if the audit tells you so. Otherwise, don't.